doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,299 @@
+
+.. _pandas:
+
+=======================
+Pandas Interoperability
+=======================
+
+The basics of using pandas and Scikit-learn
+==================================================================
+
+In principle, Scikit-learn supports the use of
+`pandas DataFrames <http://pandas.pydata.org/pandas-docs/stable/>`__
+implicitly. However, this implicit support has its conditions and potential
+pitfalls and some (not all) of these will be outlined below.
+
+Reasons for these conditions and the difficulty of supporting pandas, stems
+from the fact that pandas data structures such as Series and DataFrames can
+hold heterogenous datatypes (different columns can contain different
+datatypes). Therefore, Series and DataFrames can be thought of as containers
+for arrays that hold the actual data in an homogenous format. Some of these
+underlying data structures might not necessarily be representable as NumPy
+arrays (e.g. Categorical data). Pandas supports these alternate data types using
+pandas specific or 3rd party libraries that extend the NumPy type system. The
+difficulty of supporting pandas data structure thus comes from pandas
+capability of supporting heterogenous data in one data structure.
+
+Every Scikit-learn estimator supports the use of DataFrames which is achieved
+by obtaining a `NumPy array <https://docs.scipy.org/doc/numpy/user/>`__ using
+the :meth:`.to_numpy` property of the DataFrame class.
+
+.. note::
+  Starting with pandas version 0.24.0, it is encouraged to obtain
+  NumPy arrays from DataFrames or Series via :meth:`.to_numpy()` instead of
+  using :meth:`.values`. More details on this can be found in the
+  `release notes <http://pandas-docs.github.io/pandas-docs-travis/whatsnew/v0.24.0.html#accessing-the-values-in-a-series-or-index>`__
+  and the documentation `here <http://pandas.pydata.org/pandas-docs/stable/getting_started/basics.html#basics-dtypes>`__
+  and `here <http://pandas.pydata.org/pandas-docs/stable/getting_started/basics.html#attributes-and-underlying-data>`__.
```

Comment:
I am happy to remove the section but as per discussion above I got the feeling that mentioning it in a small note might be worth. After all it could be that some scikit-learn users that are also pandas users will hear about it this way because there are not following all the pandas updates. This could be appreciated by some.

Reply:
I agree this is not relevant.

Reply:
removed